New OpenFisca website <§>

clementbiron opened the pull request:
This PR contains the new Hugo theme for OpenFisca.

Try it localy with `hugo server -w -v --disableFastRender` and browse through `http://localhost:1313/en/`.

![image](https://user-images.githubusercontent.com/364319/210511832-8adbecda-9a73-4f7d-a619-0f2b6e85eda9.png)

MattiSG commented on themes/OpenFisca-2022/layouts/_default/packages.html:
```diff
@@ -0,0 +1,77 @@
+    <body>
+        {{ partial "sprite.svg" . }}
+
+        {{ partial "header.html" . }}
```
Please [inherit from baseof](https://gohugo.io/templates/base/#override-the-base-template) for maintainability :)

MattiSG commented on themes/OpenFisca-2022/layouts/_default/packages.html:
```diff
@@ -0,0 +1,77 @@
+                            <tr>
+                                <td data-column="{{ i18n "packages.jurisdiction" }}">
+                                    {{ if eq .jurisdiction "ZZ" }}
+                                        <span style="color:var(--colorGray500)">…</span>
```
```suggestion
                                        <span style="color:var(--colorGray500)">🏳️</span>
```
…or the `<§>` small logo? 😃

MattiSG commented on themes/OpenFisca-2022/layouts/_default/packages.html:
```diff
@@ -0,0 +1,77 @@
+                                    {{ with .legislation_explorer }}
+                                        <a href="{{ . }}" target="_blank">{{ . }}</a>
+                                    {{ else }}
+                                        <span style="color:var(--colorGray500)">…</span>
```
These ellipses are a bit confusing. A cross or empty?

MattiSG commented on themes/OpenFisca-2022/layouts/_default/showcase.html:
```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE html>
+<html lang="{{ $.Site.Language }}">
+
+    {{ partial "head.html" . }}
```
Inherit from `baseof` 🙂

MattiSG commented on themes/OpenFisca-2022/layouts/index.html:
```diff
@@ -0,0 +1,70 @@
+         </div>
+      </div>
+   </div>
+
+{{ end }}
```
```suggestion
{{ end }}
```

clementbiron replied:
Sorry, I don't understand and this doesn't correspond to anything I noted during our peer session. Can you tell me more? Thanks.

MattiSG replied:
Just suggesting to remove the empty blank line 😉

MattiSG commented on themes/OpenFisca-2022/layouts/partials/footer.html:
```diff
@@ -0,0 +1,26 @@
+<footer class="footer">
+	<div class="footer_content">
+		<div class="footer_baseline"><§> #RulesAsCode</div>
```
Please make sure to encode these HTML signs throughout the codebase!
```suggestion
		<div class="footer_baseline">&lt;§&gt; #RulesAsCode</div>
```

MattiSG commented on themes/OpenFisca-2022/layouts/partials/footer.html:
```diff
@@ -0,0 +1,26 @@
+</footer>
+
+{{ $ludcideJs := resources.Get "js/lucide.js" | js.Build }}
+<script src="{{ $ludcideJs.RelPermalink }}"></script>
```
```suggestion
<script src="{{ $lucideJs.RelPermalink }}"></script>
```

MattiSG commented on themes/OpenFisca-2022/layouts/partials/head.html:
```diff
@@ -0,0 +1,21 @@
+<head>
+	<meta charset="utf-8" />
+	<title>OpenFisca - {{ .Title}}</title>
```
```suggestion
	<title>{{ .Title}} — OpenFisca</title>
```

MattiSG commented on themes/OpenFisca-2022/theme.toml:
```diff
@@ -0,0 +1,6 @@
+name = "OpenFisca 2022"
+license = "MIT"
+licenselink = "https://github.com/openfisca/openfisca.org/themes/OpenFisca-2022//blob/master/LICENSE"
```
```suggestion
licenselink = "https://github.com/openfisca/openfisca.org/themes/OpenFisca-2022/blob/master/LICENSE"
```

MattiSG commented on themes/OpenFisca-2022/theme.toml:
```diff
@@ -0,0 +1,6 @@
+name = "OpenFisca 2022"
+license = "MIT"
+licenselink = "https://github.com/openfisca/openfisca.org/themes/OpenFisca-2022//blob/master/LICENSE"
+description = "The most widely adopted free and open-source engine to write rules as code"
+homepage = "https://openfisca.org"
```
This is actually not the description of the theme, and it could be confusing. I suggest we correct (or remove) it 🙂

MattiSG reviewed (changes requested):
This is amazing 🤩 

- Remove the unused `OpenFisca-2018` theme.
- Replace paletted JPEG images with PNG (preferably with a fixup) to decrease storage and bandwidth requirements 🌱
- A CSS linter would be very welcome to unify spaces (one before `{` and after `:`).
- Remove empty `main.css` file.
- Run ESlint on the JS files.
- Document how to add icons in README.
- Document `styleguide` in README.

clementbiron commented on .circleci/config.yml:
```diff
@@ -62,8 +61,12 @@ workflows:
           html-proofer: false
+          pre-steps:
+            - checkout
+            - node/install
```
I have not found any other way to install npm dependencies with `circleci/hugo@1.3.0`.

MattiSG commented on README.md:
````diff
@@ -118,3 +118,19 @@ For development purposes:
+To lint the JavaScript files use the following command:
+
+```sh
+npm run lint:js --fix
````
Are you sure this works? I believe it should be `npm run lint:js -- --fix` to pass the `fix` option to the linter 🙂

I also feel like it would be more appropriate to not document `fix` as the default behaviour. If we really believe the best is to always auto fix, then we should add the option in the script itself.

```suggestion
npm run lint:js
```

MattiSG commented on README.md:
````diff
@@ -118,3 +118,19 @@ For development purposes:
+To lint the CSS files use the following command:
+
+```sh
+npm run lint:css --fix
````
See above 
```suggestion
npm run lint:css
```

MattiSG reviewed (commented):
This is fantastic 🤩 

Congratulations for the work on editing the commits, they are beautifully crafted and this secondary review was a breeze 🙇

The only blocking point left is a copy review / improvement that falls unto me 🙂

clementbiron commented:
To browse the 404 page, please note:

> hugo server **will not automatically load your custom 404.html file**, but you can test the appearance of your custom “not found” page by navigating your browser to /404.html.

MattiSG reviewed (commented):
We're almost there!

Just a few points:

- Why are the intermediate titles of the home page defined in `i18n/$lang.yaml` and not in `content/_index.$lang.md` like all the other content?
- Also, I feel like there is a duplicate footer navigation between `i18n/$lang.yaml` and `_default/menus.$lang.md`.
- There is a space before the Australian flag, which makes it out of place with the other flags on the Packages page.

MattiSG commented:
I see that we removed `CNAME`, but I cannot find how it is set up now. Can you clarify how that file is generated to ensure the domain will still be mapped properly? 🙂

MattiSG commented:
Oh, we also need to make sure that https://openfisca.org/img/logo-openfisca.svg still hosts a valid image! Several reusers hotlink to it.

clementbiron commented:
> There is a space before the Australian flag, which makes it out of place with the other flags on the Packages page.

![image](https://user-images.githubusercontent.com/364319/213152863-9fffd8b1-fd15-4dd4-a6ff-b508ae2f31a5.png)


I do not see this behaviour on Firefox and Chrome (MacOs, latest version), it's seems to be a Safari bug. I propose that we open an RP to deal with this later.

clementbiron commented:
> I see that we removed CNAME, but I cannot find how it is set up now. Can you clarify how that file is generated to ensure the domain will still be mapped properly? 🙂

Well catched ! The CNAME file was moved and deleted, i made a fixup.

clementbiron commented:
> Oh, we also need to make sure that https://openfisca.org/img/logo-openfisca.svg still hosts a valid image! Several reusers hotlink to it.

Ok i made a fixup and updated the logo.

MattiSG commented:
- “Try the web API” links to https://legislation.fr.openfisca.org/swagger when it should link to https://legislation.demo.openfisca.org/swagger
- “Try the Python API” links to https://legislation.fr.openfisca.org/swagger when it should link to an online demo; if we did not agree on any specific better one, then let's use https://openfisca.org/doc/installation/howto-web-no-local-install.html

MattiSG reviewed (approved):
The comments above are important to fix, but are not blocking as long as they are fixed soon, and I will not be able to review again this afternoon. I thus approve this PR and let you fix the links, rebase and merge 🙂 

Let's ship!! 🚢 🚀  …and fix the issues as they will come 😉 This is an amazing update!
